src/views/InteractiveMap/InteractiveMap.vue

```diff
@@ -457,59 +457,95 @@ export default Vue.extend({
           hasSimulationError: false
         }
       });
-      axios
-        .post(`${settings.apis.model}/simulate`, {
-          model_id: model.id,
-          method: card.method,
-          operations: operations,
-          objective_id: card.objective.reaction
-            ? card.objective.reaction.id
-            : model.default_biomass_reaction,
-          objective_direction: card.objective.maximize ? "max" : "min"
-        })
-        .then(response => {
-          const wildtypeFluxdistribution = response.data.flux_distribution;
-          // Calculate new bounds based on manipulation scores calculated by DiffFVA
-          // (reference flux * score) / production growth
-          const editedBounds = card.manipulations
-            .map(function(manipulation) {
-              const newBound = Math.round(
-                (wildtypeFluxdistribution[manipulation.id] *
-                  manipulation.value) /
-                  card.productionGrowthRate
-              );
+      // Apply the values from diffFVA results as bounds for simulation
+      const editedBounds = card.manipulations.map(function(manipulation) {
+        // do we need to filter out values that are zero or is this controlled on the backend?
+        const model_reaction = model.model_serialized.reactions.find(
+          rxn => rxn.id === manipulation.id
+        );
+        const newBound = manipulation.value;
+        switch (manipulation.direction) {
```

Comment:
In general I recommend just using a regular if-elseif statement instead of switch. In short, `switch` can cause subtle bugs when you forget `break`, or when you intentionally leave it out and later modify the code. (Douglas Crockford has a nice section on this in "Javascript: The Good Parts")

Reply:
But I didn't forget the `break` right? Isn't it save then?